### PR TITLE
adds data model for registers

### PR DESCRIPTION
**Pull request by swkeever**
Data scraped from https://en.wikichip.org/wiki/risc-v/registers

### CODE REVIEW

**Review by mcmahjoh — APPROVED** [review]

File: .babelrc
Comment (swkeever) [inline .babelrc]: maybe lets hold off on styled-jsx, it might be better to do our styles in regular CSS since we have to work also with legacy code

File: public/assets/boot.js
Comment (swkeever) [inline public/assets/boot.js]: if its not too inconvenient, could we get rid of debug prints in code before merging?

File: public/assets/elfload.js
Comment (swkeever) [inline public/assets/elfload.js]: if its not too inconvenient, could we get rid of debug prints in code before merging?

File: public/assets/elfrun.js
Comment (swkeever) [inline public/assets/elfrun.js]: you could write `if (command && command[0] === '1')` this will handle, also, the case when `command` is `undefined`

File: public/assets/elfrun.js
Comment (swkeever) [inline public/assets/elfrun.js]: same here

File: public/assets/elfrun.js
Comment (swkeever) [inline public/assets/elfrun.js]: same here

File: public/assets/elfrun.js
Comment (swkeever) [inline public/assets/elfrun.js]: same here.

also since you do the check a lot, what is the behavior if command is null? should the function throw an error? maybe we can check `if (!command)` at the beginning and then the rest of the function could assume the command is non-null

File: public/assets/webworker.js
Comment (swkeever) [inline public/assets/webworker.js]: maybe a comment in the header of `elfRunNextInst` describing the behavior when command is null or non-null. would be useful for the reader

File: src/components/RegMenu.js
Comment (swkeever) [inline src/components/RegMenu.js]: this kind of looks like a button group. maybe we could wrap in a div instead of a fragment.

see https://getbootstrap.com/docs/4.4/components/button-group/ for an example of what i mean

File: src/components/RegMenu.js
Comment (swkeever) [inline src/components/RegMenu.js]: Should we capitalize the text here in the button labels? 

What about changing the name from X-saved to just X? do you think that wouldn't be descriptive enough?

File: src/components/RegMenu.js
Comment (swkeever) [inline src/components/RegMenu.js:51]: where's caller saved filter?

should this be a switch statement instead?

File: src/components/Registers.js
Comment (swkeever) [inline src/components/Registers.js]: you don't need a fragment here

File: src/components/helpers/RegMenuHelpers.js
Comment (swkeever) [inline src/components/helpers/RegMenuHelpers.js]: what about using a template string? ```
`${reg.name} (${reg.abiName}): ${reg.value}`
```

File: src/components/helpers/RegMenuHelpers.js
Comment (swkeever) [inline src/components/helpers/RegMenuHelpers.js]: should we just define these helper functions inside of `components/Register.js`?

File: src/components/RegMenu.js
Comment (Zorrow2008) [reply src/components/RegMenu.js]: added this

File: src/components/RegMenu.js
Comment (Zorrow2008) [reply src/components/RegMenu.js:51]: added this

File: src/components/Registers.js
Comment (Zorrow2008) [reply src/components/Registers.js]: changed this

File: src/components/RegMenu.js
Comment (Zorrow2008) [reply src/components/RegMenu.js]: I agree with capitalized, I think keeping -saved is more descriptive, because I usually see the -saved is usually added along when referring to these kinds of registers, but if you think itll make the button look clunky, we can remove it

File: src/components/helpers/RegMenuHelpers.js
Comment (Zorrow2008) [reply src/components/helpers/RegMenuHelpers.js]: i think these methods will be useful, but could we add it inside helpers/Registers.js, instead of RegMenu?

File: src/components/helpers/RegMenuHelpers.js
Comment (Zorrow2008) [reply src/components/helpers/RegMenuHelpers.js]: tried this, eslint still gives the complaint about how each element needs to be on its own line

File: src/components/helpers/RegMenuHelpers.js
Comment (swkeever) [reply src/components/helpers/RegMenuHelpers.js]: sometimes its just convienent to not have to go to a separate file to access helper functions. 
but if you like them in a separate file then go ahead

File: src/components/helpers/RegMenuHelpers.js
Comment (swkeever) [reply src/components/helpers/RegMenuHelpers.js]: even if you do
```
<ReactElement>
  {`${reg.name} (${reg.abiName}): ${reg.value}`}
</ReactElement>
```

?

I don't think ESLint would complain about that. but if there's no other way around it then sure

File: src/components/RegMenu.js
Comment (swkeever) [reply src/components/RegMenu.js]: yea, how it appears on the button is all that i'm worried about, since "all" is 3 characters and "caller-saved" is >10. but we can cross that bridge we get to styling the buttons

**Pull request closed without merging**
